src/main/java/dwbh/api/services/internal/DefaultUserGroupService.java

```diff
@@ -73,63 +74,81 @@ public DefaultUserGroupService(
 
   @Override
   public Result<Boolean> addUserToGroup(AddUserToGroupInput input) {
-    Group group = groupRepository.getGroup(input.getGroupId());
-    User user = userRepository.getUserByEmail(input.getEmail());
+    Optional<Group> group = groupRepository.findById(input.getGroupId());
+    Optional<User> user = userRepository.findByEmail(input.getEmail());
 
-    NotNull notNull = new NotNull();
+    NotPresent notPresent = new NotPresent();
     UserIsGroupAdmin userIsGroupAdmin = new UserIsGroupAdmin(userGroupRepository);
     UserIsNotInGroup notInGroupChecker = new UserIsNotInGroup(userGroupRepository);
 
     return Result.<Boolean>create()
-        .thenCheck(() -> notNull.check(group))
-        .thenCheck(() -> notNull.check(user))
+        .thenCheck(() -> notPresent.check(group))
+        .thenCheck(() -> notPresent.check(user))
         .thenCheck(() -> userIsGroupAdmin.check(input.getCurrentUserId(), input.getGroupId()))
```

Comment:
Está fallando esta comprobación al tratar de abandonar un grupo.

Reproducción

1. Creo un grupo nuevo con un usuario nuevo cargado desde fixtures (Daniel)
2. Invito a otros dos usuarios (cxavier y tstark)
3. Emito tres votaciones diferentes para los tres usuarios
4. Logado como tStark visualizo mis grupos y trato de abandonar el grupo para el que acabo de votar
5. Se muestra "You can't leave this group" a pesar de no ser admin (entiendo que debería ser quien creó el grupo, Daniel), y haber más ususarios (cxavier)

`java.lang.IllegalStateException: stream has already been operated upon or closed
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:229)
        at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:543)
        at dwbh.api.services.internal.checkers.UserIsNotUniqueGroupAdmin.check(UserIsNotUniqueGroupAdmin.java:53)
        at dwbh.api.services.internal.DefaultUserGroupService.lambda$leaveGroup$9(DefaultUserGroupService.java:133)
        at dwbh.api.util.Result.thenCheck(Result.java:112)
        at dwbh.api.services.internal.DefaultUserGroupService.leaveGroup(DefaultUserGroupService.java:133)
        at dwbh.api.services.internal.$DefaultUserGroupServiceDefinition$Intercepted.$$access2(Unknown Source)
        at dwbh.api.services.internal.$DefaultUserGroupServiceDefinition$Intercepted$$proxy2.invokeInternal(Unknown Source)
        at io.micronaut.context.AbstractExecutableMethod.invoke(AbstractExecutableMethod.java:146)
        at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:60)
        at io.micronaut.transaction.interceptor.TransactionalInterceptor.intercept(TransactionalInterceptor.java:93)
        at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:69)
        at dwbh.api.services.internal.$DefaultUserGroupServiceDefinition$Intercepted.leaveGroup(Unknown Source)
        at dwbh.api.graphql.fetchers.UserGroupFetcher.leaveGroup(UserGroupFetcher.java:109)
        at graphql.execution.instrumentation.dataloader.DataLoaderDispatcherInstrumentation.lambda$instrumentDataFetcher$0(DataLoaderDispatcherInstrumentation.java:86)
        at graphql.execution.ExecutionStrategy.fetchField(ExecutionStrategy.java:270)
        at graphql.execution.ExecutionStrategy.resolveFieldWithInfo(ExecutionStrategy.java:198)
        at graphql.execution.ExecutionStrategy.resolveField(ExecutionStrategy.java:171)
        at graphql.execution.AsyncSerialExecutionStrategy.lambda$execute$1(AsyncSerialExecutionStrategy.java:41)
        at graphql.execution.Async.eachSequentiallyImpl(Async.java:77)
        at graphql.execution.Async.eachSequentially(Async.java:66)
        at graphql.execution.AsyncSerialExecutionStrategy.execute(AsyncSerialExecutionStrategy.java:36)
        at graphql.execution.Execution.executeOperation(Execution.java:161)
        at graphql.execution.Execution.execute(Execution.java:102)
        at graphql.GraphQL.execute(GraphQL.java:605)
        at graphql.GraphQL.parseValidateAndExecute(GraphQL.java:538)
        at graphql.GraphQL.executeAsync(GraphQL.java:502)
        at io.micronaut.configuration.graphql.DefaultGraphQLInvocation.lambda$null$0(DefaultGraphQLInvocation.java:69)
        at io.micronaut.core.async.publisher.CompletableFuturePublisher$CompletableFutureSubscription.request(CompletableFuturePublisher.java:76)
        at io.reactivex.internal.operators.flowable.FlowableFlatMap$InnerSubscriber.onSubscribe(FlowableFlatMap.java:656)
        at io.micronaut.core.async.publisher.CompletableFuturePublisher.subscribe(CompletableFuturePublisher.java:48)
        at io.reactivex.internal.operators.flowable.FlowableFlatMap$MergeSubscriber.onNext(FlowableFlatMap.java:163)
        at io.micronaut.reactive.rxjava2.RxInstrumentedSubscriber.onNext(RxInstrumentedSubscriber.java:68)
        at io.micronaut.core.async.publisher.Publishers$JustPublisher$1.request(Publishers.java:341)
        at io.reactivex.internal.operators.flowable.FlowableFlatMap$MergeSubscriber.onSubscribe(FlowableFlatMap.java:117)
        at io.micronaut.reactive.rxjava2.RxInstrumentedSubscriber.onSubscribe(RxInstrumentedSubscriber.java:54)
        at io.micronaut.core.async.publisher.Publishers$JustPublisher.subscribe(Publishers.java:331)
        at io.reactivex.internal.operators.flowable.FlowableFromPublisher.subscribeActual(FlowableFromPublisher.java:29)
        at io.reactivex.Flowable.subscribe(Flowable.java:14918)
        at io.reactivex.Flowable.subscribe(Flowable.java:14865)
        at io.micronaut.reactive.rxjava2.RxInstrumentedFlowable.subscribeActual(RxInstrumentedFlowable.java:58)
        at io.reactivex.Flowable.subscribe(Flowable.java:14918)
        at io.reactivex.internal.operators.flowable.FlowableFlatMap.subscribeActual(FlowableFlatMap.java:53)
        at io.reactivex.Flowable.subscribe(Flowable.java:14918)
        at io.reactivex.Flowable.subscribe(Flowable.java:14865)
        at io.micronaut.reactive.rxjava2.RxInstrumentedFlowable.subscribeActual(RxInstrumentedFlowable.java:58)
        at io.reactivex.Flowable.subscribe(Flowable.java:14918)
        at io.reactivex.Flowable.subscribe(Flowable.java:14868)
        at io.micronaut.core.async.publisher.Publishers.lambda$map$2(Publishers.java:134)
        at io.micronaut.core.async.publisher.Publishers.lambda$map$2(Publishers.java:134)
        at io.reactivex.internal.operators.flowable.FlowableDefer.subscribeActual(FlowableDefer.java:42)
        at io.reactivex.Flowable.subscribe(Flowable.java:14918)
        at io.reactivex.internal.operators.flowable.FlowableMap.subscribeActual(FlowableMap.java:37)
        at io.reactivex.Flowable.subscribe(Flowable.java:14918)
        at io.reactivex.internal.operators.flowable.FlowableSwitchIfEmpty.subscribeActual(FlowableSwitchIfEmpty.java:32)
        at io.reactivex.Flowable.subscribe(Flowable.java:14918)
        at io.reactivex.Flowable.subscribe(Flowable.java:14868)
        at io.micronaut.http.context.ServerRequestTracingPublisher.lambda$subscribe$0(ServerRequestTracingPublisher.java:52)
        at io.micronaut.http.context.ServerRequestContext.with(ServerRequestContext.java:68)
        at io.micronaut.http.context.ServerRequestTracingPublisher.subscribe(ServerRequestTracingPublisher.java:52)
        at io.micronaut.core.async.publisher.Publishers.lambda$map$2(Publishers.java:134)
        at io.reactivex.internal.operators.flowable.FlowableFromPublisher.subscribeActual(FlowableFromPublisher.java:29)
        at io.reactivex.Flowable.subscribe(Flowable.java:14918)
        at io.reactivex.Flowable.subscribe(Flowable.java:14865)
        at io.reactivex.internal.operators.flowable.FlowableSubscribeOn$SubscribeOnSubscriber.run(FlowableSubscribeOn.java:82)
        at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:288)
        at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:253)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:497)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:834)
`